### DISCUSSION TITLE
Why isnt delayed message plugin shipped with rabbitmq

### DISCUSSION THREAD

**Discussion by Snazzie** [rabbitmq/rabbitmq-delayed-message-exchange#293] [Q&A] [answered]
What is the reason for delayed plugin not being shipped with the main package, requiring manual download installation during deployment or people to package their own fork? Delayed messages feature is quite commonly used in many service architectures. and current way of enabling could be better.

**Comment by michaelklishin** [comment, marked as answer]
Because it has serious limitations very explicitly stated in the README, and it uses a data store that will be removed from RabbitMQ in a future 4.x release.

A future version of Tanzu RabbitMQ may include a delayed publishing feature but not necessarily as an exchange type and not based on this plugin. For this reason, RabbitMQ Core Team no longer works on this plugin but we do accept contributions such as #291 and #292 to keep it compatible with new releases.

There is also https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/issues/229 where a reasonably complete design outline for those who are interested in developing a replicated version.

**Comment by michaelklishin** [comment]
The process of installing this plugin is not particularly onerous:

1. Download a single file from a GitHub release
2. [Preconfigure](https://www.rabbitmq.com/docs/plugins#enabled-plugins-file) it to be enabled on boot, just like with any other plugin

In any case, the core team no longer actively works on this plugin and that won't change. This is not an essential messaging or streaming feature and the amount of effort required to rework the plugin in a way similar to #229 is too substantial to justify making the next generation open source.

We do not, however, prevent others from contributing to this plugin and as #291, #292 demonstrate, it is not completely abandoned. Since RabbitMQ is open source software, those who need this plugin
can work on it and ship it under the terms of the Mozilla Public License 2.0.
